### ISSUE TITLE
Reply By Tweet Bug Discussion

### ISSUE THREAD

**Issue by namangupta01** [publiclab/plots2#6085]
So I have figured out what the actual problem is after adding the console to print the mid way data in #6064.
The actual problem is short links.
`parent tweet #<Twitter::Tweet:0x00005646ea71df68>
parent tweet full text -> We’ve overhauled the entire https://t.co/PKr9eq2bkF website! Thanks to support from @GoogleOSS and contributors from five continents(!), we’re helping people around the world to better tell their stories through aerial mapping. Read all about it! https://t.co/nBXpX7hCHZ https://t.co/HClNdCS9y4
urls => ["https://t.co/PKr9eq2bkF", "https://t.co/nBXpX7hCHZ", "https://t.co/HClNdCS9y4"]
node => 
node.nil? => true`

![Screen Shot 2019-07-31 at 3 42 55 AM](https://user-images.githubusercontent.com/20709985/62169243-557ecf00-b345-11e9-92c1-0b1894e010bc.png)

So here, https://t.co/nBXpX7hCHZ opens to https://publiclab.org/notes/warren/07-26-2019/community-atlas-an-around-the-world-collaboration-on-mapknitter where we have implemented the algo for the https://publiclab.org/n/20331.

https://github.com/publiclab/plots2/blob/449d047f05209b443b049cdd6d3c2c905cfca51e/app/models/comment.rb#L452

I guess we can try this once with a tweet with this kind of link https://publiclab.org/n/20331
If you want to see the logs, you can see at stable.publiclab.org/cron_log1.log
What do you think?

**Comment by namangupta01** [comment]
cc @jywarren , @gauravano

**Comment by namangupta01** [comment]
Can @icarito @jywarren, please tell me what is the value of ENV["WEBSITE_HOST_PATTERN"] on stable? I need this for further debugging.

**Comment by namangupta01** [comment]
@jywarren I tried on local with the same tweet. I was able to find node on local machine but same was not happening on production. I think it is something to do with the website host pattern env variable. Can anyone of @jywarren or @icarito let me know that what is the ENV["WEBSITE_HOST_PATTERN"] set on production? Is it `//publiclab.org/n/`?

Thanks

**Comment by jywarren** [comment]
Hmm, yes it is! `export WEBSITE_HOST_PATTERN=://publiclab.org/n/`

**Comment by namangupta01** [comment]
I checked the logs and yayy, I found this:
![Screen Shot 2019-08-19 at 9 49 08 PM](https://user-images.githubusercontent.com/20709985/63282081-aee67800-c2cb-11e9-8c24-83b4d377e7bf.png)
We are able to get the node id and the only thing that I need is an email of user loggedin via twitter . But I am not getting any email for my twitter username -- I already have logged in via twitter. Its finding in the code here https://github.com/publiclab/plots2/blob/e22463991a21faf55e498b087715153f61704b1a/app/models/comment.rb#L501
Can you run this function from the rails console and find out if is there data for email : "01namangupta@gmail.com"? 
-- We are very close this time, just need to find user email.

Thanks!!

**Comment by namangupta01** [comment]
Also, can you delete the logs file or copy it somewhere else /public/cron_log1.log file on publiclab. Because this file is big and I need to download it very ofter after doing various reply by tweet.
The size is getting big is because of we are printing this data https://github.com/publiclab/plots2/blob/e22463991a21faf55e498b087715153f61704b1a/app/models/comment.rb#L509.

**Comment by jywarren** [comment]
Hi, @namangupta01 what query exactly should I run or what data should I look for? `Comment. find_email()` for your twitter handle?

**Comment by jywarren** [comment]
If it's helpful, I've just logged in via twitter! https://publiclab.org/profile/warren/

**Comment by namangupta01** [comment]
Yes! And also can you also try to run with your username as well?

**Comment by namangupta01** [comment]
Can you run it with two of my usernames?
1. itsmenamangupta
2. NamanGu78470041

**Comment by jywarren** [comment]
`Comment.find_email('itsmenamangupta')` returned `nil`, and `Comment.find_email('NamanGu78470041')` returned "naman18996@gmail.com"

**Comment by namangupta01** [comment]
Yes, this is the problem. This is what I was thinking. 
The problem is the people who logged in before we started storing for users data in UserTag if not present there.

**Comment by jywarren** [comment]
Ah, ok. Do we need to re-prompt them somehow? Or could we run a query to
generate a list of people for whom this is the case?

**Comment by namangupta01** [comment]
I will look to the flow of login via twitter and will add condition there
to save the data in UserTag whenever any user login again via twitter.

**Comment by namangupta01** [comment]
And also can you delete the file /cron_log1.log? Its is getting very big?

**Comment by jywarren** [comment]
Ah that's great. If we could generate the list we could also ask people to
log in once again.

**Comment by namangupta01** [comment]
Yup we can do!

**Comment by jywarren** [comment]
Ah yes, will delete it now.

**Comment by namangupta01** [comment]
Let me know when its deleted. I will check with the new tweet made by
naman18996 account.

**Comment by jywarren** [comment]
Its deleted!

**Comment by namangupta01** [comment]
Yay, It's working!!! 🎉 🎉 
![Screen Shot 2019-08-20 at 11 23 16 AM](https://user-images.githubusercontent.com/20709985/63321022-15f04500-c33d-11e9-970a-e8a7660394ee.png)
Link here: https://publiclab.org/notes/mimiss/08-05-2019/trawling-with-babylegs-in-new-orleans-city-park
This was done by reply by tweet.

LOL, this was working all this time the only thing is that we were not doing reply by a tweet from the proper accounts. By proper accounts, I mean those accounts whose first login via twitter was done after we implemented this feature --as I said above.
Finally, it's working just have to solve the case with old accounts :)

**Comment by jywarren** [comment]
Oh wow! That's great! So, we still have to be sure it uses a proper /n/____
shortcode, not a t.co one, right?

Can we add a "this comment was added by tweet" and corresponding icon to
this template?

https://github.com/publiclab/plots2/blob/3a0e9744d626ebfa7646ab2b406b8244172b6468/app/views/notes/_comment.html.erb#L52-L54

This is amazing - it's working in production?

**Comment by namangupta01** [comment]
Yes! This is working in production.

**Comment by namangupta01** [comment]
Here's the tweet where I replied https://twitter.com/PublicLab/status/1159583447926677505

**Comment by namangupta01** [comment]
In this tweet, it uses /n/___. But I am not sure about t.co but we can also test this out as well. And sure I will add the icon.

**Comment by jywarren** [comment]
Great! I tweeted here: https://twitter.com/jywarren/status/1163852083747012609 but maybe mine is an old account?

**Comment by namangupta01** [comment]
If Comment.find_email is able to find email from your user name then you can.

**Comment by jywarren** [comment]
Ah, it did find an email but it wasn't the email I am signed up on PL for,
is that OK?

Also, how often does it check? Is it possible i need to wait longer?

**Comment by namangupta01** [comment]
This is weird. If you have logged in via twitter then your twitter email and email on PublicLab is supposed to be same? Isn't it?

**Comment by namangupta01** [comment]
I got this here in the logs: 
![Screen Shot 2019-08-20 at 11 43 51 PM](https://user-images.githubusercontent.com/20709985/63372783-78306080-c3a4-11e9-957f-9b1799966e1b.png)

**Comment by namangupta01** [comment]
Can you tell me how different email id case?

**Comment by namangupta01** [comment]
Oh, I got it.

**Comment by namangupta01** [comment]
Here's the fix jeff https://github.com/publiclab/plots2/pull/6166

**Comment by jywarren** [comment]
Hi @namangupta01 do you think you could remove the comments used in reply by tweet now?

**Comment by namangupta01** [comment]
Sure!

**Comment by jywarren** [comment]
Sorry about the stalebot message here, it was a mistake! 😅
